core/src/validator.rs

```diff
@@ -659,35 +680,36 @@ fn backup_and_clear_blockstore(blockstore: &Arc<Blockstore>, ledger_path: &Path,
                 );
                 last_print = Instant::now();
             }
+            last_slot = Some(slot);
         }
 
-        info!("Purging slots {} to {}", start_slot, end_slot);
-        blockstore.purge_slots_with_delay(start_slot, end_slot, None, PurgeType::Exact);
-        info!("Purging done, compacting db..");
-        if let Err(e) = blockstore.compact_storage(start_slot, end_slot) {
-            warn!(
-                "Error from compacting storage from {} to {}: {:?}",
-                start_slot, end_slot, e
-            );
+        if let Some(end_slot) = last_slot {
+            info!("Purging slots {} to {}", start_slot, end_slot);
+            blockstore.purge_slots_with_delay(start_slot, end_slot, None, PurgeType::Exact);
+            blockstore.purge_from_next_slots(start_slot, end_slot);
```

Comment:
Double checking, `purge_from_next_slots` is inclusive of the `end_slot`?

Reply:
yup they are 😄

Reply:
great.